model_dealership.rb

```diff
@@ -0,0 +1,68 @@
+require 'csv'
+
+module CarLoader
+  def self.get_cars_from_csv(filepath)
+    @cars = CSV.read(filepath, { :headers => true, :header_converters => :symbol}).map { |car_object| Car.new(car_object) }
+  end
+end
+
+class Car
+  include CarLoader
+  attr_reader :make, :year, :model, :inventory_number
+  def initialize(args)
+    @inventory_number = args[:inventory_number]
+    @make = args[:make]
+    @model = args[:model]
+    @year = args[:year]
+  end
+
+  def to_s
+    "#{self.year} #{self.make} #{self.model} ID: #{self.inventory_number}" + "\n"
+  end
+end
+
+class Dealership
+  attr_accessor :cars
+  def initialize(cars = nil)
+    @cars = cars || []
+  end
+
+  def save
+    CSV.open('inventory.csv',"w") do |csv|
+      cars.each do |car|
+        csv << [car.to_s]
```

Comment:
`[car.to_s]` is not sufficient here. Before this each loop, you need to insert the headers into the CSV file. Then as you loop through and push rows in, each row should be an array with as many elements as their are columns.